chapter/related-work.md

```diff
@@ -6,6 +6,90 @@ As such, it is important to get a picture of the field of current LSP projects.
+| intellij-community/python | 47720 (C) + 248177 (Java) + 37101 (Kotlin) + 277125 (Python) |
+
+: Comparison of the size for different IntelliJ platform plugins {#tbl:plugins-size}
+
+Naturally, development efforts at this size would gravitate around the most promising solution, stifling the progress on competing platforms [@intellij-comparison-eclipse].
+Additionally, it would lock-in programmers into a specific platform for its language support regardless of their personal preference.
+
+### IDE Abstraction
+
+#### Monto
+
+The authors of the Monto project[@monto,@monto-disintegrated] call this the "IDE Portability Problem".
+They compare the situation with the task of compiling different high level languages to a set of CPU architectures.
+The answer to that problem was an intermediate representation (IR).
+Compilers could transform input languages into this IR and in turn generate assembly for different architectures from a single input format.
+
+With Monto, Kreidel et al propose a similar idea for IDE portability.
+The paper describes the *Monto IR* and how they use a *Message Broker* to receive events from the Editor and dispatch them to *Monto Services*.
+
+The Monto IR is a language agnostic and editor independent, JSON serialized tree-like model.
+Additionally, the IR maintains low level syntax highlighting information (font, color, style, etc.) but leaves the highlighting to the language specific service.
+
+The processing and modification of the source code and IR is performed by *Monto Services*.
+Services implement specific actions, e.g. parsing, outlining or highlighting.
+A central broker connects the services with each other and the editor.
+
+Since Monto performs all work on the IR, independent of the editor, and serializes the IR as JSON messages, the language used to implement *Monto Services* can be chosen freely giving even more flexibility.
+
+The Editor extension's responsibility is to act as a source and sink for data.
+It sends Monto compliant messages to the broker and receives processing results such as (error) reports.
+The communication is based on the ZeroMQ[zeromq] technology which was chosen because it is lightweight and available in manly languages [@monto-disintegrated] allowing to make use of existing language tools.
+
+#### Merlin
+
+The Merlin tool [@merlin,@merlin-website] is in many ways a more specific version of the idea presented in Monto.
+Merlin is a language server for the Ocaml language, yet predates the Language Server Protocol.
+
+The authors of Merlin postulate that implementing "tooling support traditionally provided by IDEs" for "niche languages" demands to "share the language-awareness logic" between implementations.
+As an answer to that, they describe the architecture of Merlin in [@merlin].
+
+Similarly to Monto, Merlin separates editor extensions from language analysis.
+Conversely, its interaction builds on a command line interface instead of message passing.
```

Comment:
```suggestion
However, Merlin uses a command line interface instead of message passing for interaction.
```
Just a suggestion, but `conversely` sounded a bit out of place here.